core/trino-spi/src/main/java/io/trino/spi/function/ScalarFunctionAdapter.java

```diff
@@ -190,7 +190,7 @@ public MethodHandle adapt(
         requireNonNull(methodHandle, "methodHandle is null");
         requireNonNull(actualConvention, "actualConvention is null");
         requireNonNull(expectedConvention, "expectedConvention is null");
-        if (expectedConvention.getArgumentConventions().size() != expectedConvention.getArgumentConventions().size()) {
+        if (actualConvention.getArgumentConventions().size() != expectedConvention.getArgumentConventions().size()) {
```

Comment:
That one is actually interesting

Reply:
I assume this currently has no end-user visible impact and i won't write release notes for this.
@dain  please correct me if I am wrong here.